torchaudio/transforms.py

```diff
@@ -233,6 +234,79 @@ def forward(self, specgram):
         return mel_specgram
 
 
+class InverseMelScale(torch.nn.Module):
+    r"""Solve for a normal STFT from a mel frequency STFT, using a conversion
+    matrix.  This uses triangular filter banks.
+
+    It minimizes the euclidian norm between the input mel-spectrogram and the product between
+    the estimated spectrogram and the filter banks using SGD.
+
+    Args:
+        n_stft (int): Number of bins in STFT. See ``n_fft`` in :class:`Spectrogram`.
+        n_mels (int): Number of mel filterbanks. (Default: ``128``)
+        sample_rate (int): Sample rate of audio signal. (Default: ``16000``)
+        f_min (float): Minimum frequency. (Default: ``0.``)
+        f_max (float, optional): Maximum frequency. (Default: ``sample_rate // 2``)
+        max_iter (int): Maximum number of optimization iterations.
+        tolerance_loss (float): Value of loss to stop optimization at.
+        tolerance_change (float): Difference in losses to stop optimization at.
+        sgdargs (dict): Arguments for the SGD optimizer.
+    """
+    __constants__ = ['n_stft', 'n_mels', 'sample_rate', 'f_min', 'f_max', 'max_iter', 'tolerance_loss',
+                     'tolerance_change', 'sgdargs']
+
+    def __init__(self, n_stft, n_mels=128, sample_rate=16000, f_min=0., f_max=None, max_iter=100000,
+                 tolerance_loss=1e-5, tolerance_change=1e-8, sgdargs=None):
+        super(InverseMelScale, self).__init__()
+        self.n_mels = n_mels
+        self.sample_rate = sample_rate
+        self.f_max = f_max or float(sample_rate // 2)
+        self.f_min = f_min
+        self.max_iter = max_iter
+        self.tolerance_loss = tolerance_loss
+        self.tolerance_change = tolerance_change
+        self.sgdargs = sgdargs or {'lr': 0.1, 'momentum': 0.9}
+
+        assert f_min <= self.f_max, 'Require f_min: %f < f_max: %f' % (f_min, self.f_max)
+
+        fb = F.create_fb_matrix(n_stft, self.f_min, self.f_max, self.n_mels, self.sample_rate)
+        self.register_buffer('fb', fb)
+
+    def forward(self, melspec):
+        r"""
+        Args:
+            melspec (torch.Tensor): A Mel frequency spectrogram of dimension (channel, ``n_mels``, time)
```

Comment:
Could we support batching by doing `view` as in other transforms?

Reply:
@vincentqb `channel` here is effectively the batch dimension?

Reply:
Added support for arbitrary batch dimensions. https://github.com/pytorch/audio/pull/366/commits/590939e07327d13a854bcdccac77a93ebac1a95d

Reply:
Can you also add a test as done [here](https://github.com/pytorch/audio/blob/master/test/test_transforms.py#L329) for this?

Reply:
(just saw you added tests as done [here](https://github.com/pytorch/audio/blob/master/test/test_transforms.py#L329), thanks!)

Reply:
Thanks @jaeyeun97. I incorporated your commit and made small changes for Python2 compatibility. (just in case)
Also added batch test.

Reply:
Most transforms are reshaped using view to fold the batches into the channel dimensions, and then unfolded, see [here](https://github.com/pytorch/audio/blob/master/torchaudio/transforms.py#L216).

Reply:
@vincentqb Will do so.